File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
from itertools import chain

def solution(n):
    [x, y, num] = [-1, 0, 1]
    board = [[None] * i for i in range(1, n+1)]
    
    for i in range(n):
        for _ in range(n-i):
            if i % 3 == 0: x += 1
            elif i % 3 == 1: y += 1
            else:
                x -= 1
                y -= 1
            board[x][y] = num
            num += 1
            
    return list(chain(*board))

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.js
function solution(n) {
  let tri = Array(n).fill().map((_, i) => Array(i + 1).fill())
  let x = -1, y = 0
  let num = 0
  
  for (let i = n; i > 0; i -= 3) {
    tri[++x][y] = ++num
    for (let j = 0; j < i - 1; j++) tri[++x][y] = ++num
    for (let j = 0; j < i - 1; j++) tri[x][++y] = ++num
    for (let j = 0; j < i - 2; j++) tri[--x][--y] = ++num
  }
  return tri.flat()
}